discovery-provider/src/tasks/users.py

```diff
@@ -188,27 +190,56 @@ def parse_user_event(
+        # Look up metadata multihash in IPFS and override with metadata fields
+        ipfs_metadata = get_ipfs_metadata(
+            update_task, user_record
+        )
+
+        if ipfs_metadata:
+            # ipfs_metadata properties are defined in get_ipfs_metadata
+
+            # Fields also stored on chain
+            if "profile_picture" in ipfs_metadata and \
+                ipfs_metadata["profile_picture"]:
+                user_record.profile_picture = ipfs_metadata["profile_picture"]
+
+            if "cover_photo" in ipfs_metadata and \
+                ipfs_metadata["cover_photo"]:
+                user_record.cover_photo = ipfs_metadata["cover_photo"]
+
+            if "bio" in ipfs_metadata and \
+                ipfs_metadata["bio"]:
+                user_record.bio = ipfs_metadata["bio"]
+
+            if "name" in ipfs_metadata and \
+                ipfs_metadata["name"]:
+                user_record.name = ipfs_metadata["name"]
+
+            if "location" in ipfs_metadata and \
+                ipfs_metadata["location"]:
+                user_record.location = ipfs_metadata["location"]
+
+            # Fields with no on-chain counterpart
+            if "profile_picture_sizes" in ipfs_metadata and \
+                ipfs_metadata["profile_picture_sizes"]:
+                user_record.profile_picture = ipfs_metadata["profile_picture_sizes"]
+
+            if "cover_photo_sizes" in ipfs_metadata and \
+                ipfs_metadata["cover_photo_sizes"]:
+                user_record.cover_photo = ipfs_metadata["cover_photo_sizes"]
+
+            if "collectibles" in ipfs_metadata and \
+                ipfs_metadata["collectibles"] and \
+                isinstance(ipfs_metadata["collectibles"], dict) and \
+                ipfs_metadata["collectibles"].items():
+                user_record.has_collectibles = True
+            else:
+                user_record.has_collectibles = False
 
-    if metadata_overrides:
-        # metadata_overrides properties are defined in get_metadata_overrides_from_ipfs
-        if metadata_overrides["profile_picture"]:
-            user_record.profile_picture = metadata_overrides["profile_picture"]
-        if metadata_overrides["profile_picture_sizes"]:
-            user_record.profile_picture = metadata_overrides["profile_picture_sizes"]
-        if metadata_overrides["cover_photo"]:
-            user_record.cover_photo = metadata_overrides["cover_photo"]
-        if metadata_overrides["cover_photo_sizes"]:
-            user_record.cover_photo = metadata_overrides["cover_photo_sizes"]
-        if metadata_overrides["bio"]:
-            user_record.bio = metadata_overrides["bio"]
-        if metadata_overrides["name"]:
-            user_record.name = metadata_overrides["name"]
-        if metadata_overrides["location"]:
-            user_record.location = metadata_overrides["location"]
+            if 'associated_wallets' in ipfs_metadata:
```

Comment:
very minor question, should an empty dict be passed in if for some reason there previouslly was associated_wallets and now there are not? not 100 if that is even posssiible (idt so) with happy pathclient

Reply:
in the metadata object i mean

Reply:
the json schema should nullify it, so we should always have something here